lispyville.el

```diff
@@ -665,6 +665,74 @@ ARG has the same effect."
       (lispy--normalize-1))
     (goto-char orig-pos)))
 
+(evil-define-operator lispyville-comment-or-uncomment (beg end type)
+  "Like `evilnc-comment-operator' but handles unmatched delimiters.
+May insert newlines where necessary in order to move delimiters outside of the
+commented regions and preserve the balanced structure. "
+  (interactive "<R>")
+  (when (string-match-p (format "[%s]+$" (regexp-quote comment-start))
+                        (buffer-substring beg end))
+    (setq end (point-at-eol)))
+  (let* ((end-marker (copy-marker (1+ end))))
+    (save-excursion
+      (dolist (r (lispy--find-safe-regions beg end))
+        (comment-or-uncomment-region
+         (car r)
+         ;; must include trailing whitespace to avoid error with undo-tree
+         (progn (goto-char (cdr r))
+                (skip-chars-forward "[:space:]") (point)))))
+    (delete-trailing-whitespace beg (marker-position end-marker)))
+  (if (eq type 'line) (back-to-indentation)))
+
+(defun lispyville--join-between-comments ()
+  "Insert the contents of the next line between the non-commented
+and commented regions of the current line.
+If both lines have inline comments, place them after another in their original order,
+and do not strip the comment characters."
+  (let ((inline-comment
+         (delete-and-extract-region
+          (save-excursion (forward-line 0)
+                          (search-forward comment-start (point-at-eol) 'no-error)
+                          (backward-char 1)
+                          (point))
+          (point-at-eol))))
+    (join-line 1)
+    (indent-according-to-mode)
+    (if (search-forward comment-start (point-at-eol) 'no-error)
+        (skip-chars-backward (concat "[:space:]" (regexp-quote comment-start))))
+    (save-excursion
+      (insert " " inline-comment))))
+
+(evil-define-operator lispyville-join (beg end)
+  "Join the next line keeping inline comments out of the way.
+If region is active, join all selected lines into one and splice all comments to the end."
+  :motion evil-line
+  (let ((count (count-lines beg end)))
+    (when (> count 1)
+      (setq count (1- count)))
+    (goto-char beg)
+    (dotimes (_ count)
+      (if (save-excursion (end-of-line) (lispy--in-comment-p))
+          ;; Current line is a comment or ends in an inline comment
+          (cond
+           ;; Edge case: end of buffer, do nothing
+           ((= (point-at-eol) (point-max)) nil)
+           ;; Entire next line is also a comment - splice them together
+           ((save-excursion (forward-line 1) (back-to-indentation) (lispy--in-comment-p))
+            (delete-region (point-at-eol)
+                           (progn (forward-line 1)
+                                  (skip-chars-forward (concat "[:space:]" (regexp-quote comment-start)))
```

Comment:
Skip over `comment-start-skip` instead (e.g. use `looking-at` and `match-end` if it is succesful).